### PR TITLE
ddl: we shouldn't make a DDL job to a dist task when it's on system DB in upgrading

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/48706

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
Ran IT as described in the issue
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by ywqzzy — APPROVED** [review]

File: pkg/ddl/index.go [hunk 1]
```diff
@@ -771,22 +771,28 @@ func pickBackfillType(ctx context.Context, job *model.Job, unique bool, d *ddlCt
...
 			}
 			var pdLeaderAddr string
+			var isUpgradingSysDB bool
 			if d != nil {
 				//nolint:forcetypeassert
 				pdLeaderAddr = d.store.(tikv.Storage).GetRegionCache().PDClient().GetLeaderAddr()
+				isUpgradingSysDB = d.stateSyncer.IsUpgradingState() && hasSysDB(job)
 			}
-			if variable.EnableDistTask.Load() {
+			isDistReorg := false
```
Comment (Defined2014) [inline pkg/ddl/index.go:780]: Maybe `useDistReorg` is better to understand?

File: pkg/ddl/index.go [hunk 2]
```diff
@@ -771,22 +771,28 @@ func pickBackfillType(ctx context.Context, job *model.Job, unique bool, d *ddlCt
...
+			if variable.EnableDistTask.Load() && !isUpgradingSysDB {
 				_, err = ingest.LitBackCtxMgr.Register(ctx, unique, job.ID, d.etcdCli, pdLeaderAddr, job.ReorgMeta.ResourceGroupName)
+				isDistReorg = true
 			} else {
 				_, err = ingest.LitBackCtxMgr.Register(ctx, unique, job.ID, nil, pdLeaderAddr, job.ReorgMeta.ResourceGroupName)
+				if isUpgradingSysDB {
+					logutil.BgLogger().Info("pick backfill type, handle the job on the system DB in upgrading state",
```
Comment (Defined2014) [inline pkg/ddl/index.go:787]: The log is harder to understand. :(

File: pkg/ddl/index.go (hunk 1 above)
Comment (Defined2014) [reply pkg/ddl/index.go:780]: Or no need this variable, just move Line795 to Line783

**Review by Defined2014 — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**